modules/apps/static/portal-dependency-manager-component-executor-factory/portal-dependency-manager-component-executor-factory/src/main/java/com/liferay/portal/dependency/manager/component/executor/factory/internal/DependencyManagerSyncImpl.java

```diff
@@ -135,10 +144,52 @@ public void complete(Future<Void> future) {
 			});
 	}
 
+	private void _registerDependencyManagerSyncGatekeeperCallable() {
+		registerSyncCallable(
+			() -> {
+				String componentName =
+					DependencyManagerSyncGatekeeper.class.getName();
+
+				CountDownLatch countDownLatch = new CountDownLatch(1);
+
+				BundleContext bundleContext =
+					_componentContext.getBundleContext();
+
+				ServiceListener serviceListener = serviceEvent -> {
+					if (serviceEvent.getType() != ServiceEvent.REGISTERED) {
+						return;
+					}
+
+					bundleContext.getService(
+						serviceEvent.getServiceReference());
+
+					countDownLatch.countDown();
+				};
+
+				bundleContext.addServiceListener(
+					serviceListener, "(&(objectClass=" + componentName + "))");
+
+				_componentContext.enableComponent(componentName);
+
+				while (true) {
+					if (countDownLatch.await(1, TimeUnit.MINUTES)) {
```

Comment:
@yuhai Is it necessary to wait for the `countDownLatch` every 1 min in a infinite loop? We should be able to just use the `await()` method, which will block the current thread until `countDown()`.

Reply:
@dantewang, I just push force the SF commit to change this.